### PR TITLE
[Frontend][PyTorch][Bugfix][Feature] Add support for common Pytorch operations (repeat, batchnorm1d, log_softmax)

### PR DESCRIPTION
**Pull request by ZRQ0410**
## Description ##
This PR fixes cornell-zhang/allo#386 and enhances the PyTorch frontend in the allo library to support additional operations commonly used in Transformer models.

### Problems ###
The current PyTorch frontend lacks support for several commonly used Transformer components, leading to conversion failures when compiling models.
Unsupported features include:
- `build_get_attr` handling for cls_token to avoid attribute access errors
- `repeat` for CLS token expansion
- `log_softmax` operator
- `BatchNorm1d` supporting both 2D (B, C) and 3D (B, C, L) inputs
- `relu3d` (previously only relu2d and relu4d existed)
- `concat` for sequence concatenation, used to prepend CLS token
- Slice operation to extract CLS token (Allo currently does not support slicing)

### Proposed Solutions ###
- Fix bug: add build_get_attr support for cls_token parameters
- Add repeat_batch3d, concat, relu3d, log_softmax, batchnorm1d_2d, batchnorm1d_3d and schedules
- Add leaf module SliceClsToken_lib for extracting CLS token

### Examples ###
A minimal working Particle Transformer example is provided in `examples/torch/particle_transformer.py`.

## Checklist ##

Please make sure to review and check all of these items:
- [x] PR's title starts with a category (e.g. [Bugfix], [IR], [Builder], etc)
- [x] All changes have test coverage (It would be good to provide ~2 different test cases to test the robustness of your code)
- [x] Pass the [formatting check](https://cornell-zhang.github.io/allo/developer/index.html#id1) locally
- [x] Code is well-documented

### CODE REVIEW

File: allo/frontend/library.py [hunk 1]
```diff
@@ -45,3 +45,15 @@ def CoreAttention(
         return Attn
 
     return CoreAttention
+
+
+def SliceClsToken_lib(s_0, s_1, s_2):
```
Comment (chhzh123) [inline allo/frontend/library.py:50]: Could you add a comment block on what this function is used for? You could add any references if you have

File: allo/frontend/pytorch.py [hunk 2]
```diff
@@ -344,6 +344,7 @@ def build(self):
 
     def __call__(self, node):
         method = getattr(self, "build_" + node.op)
+
```
Comment (chhzh123) [inline allo/frontend/pytorch.py]: Remove

**Review by chhzh123 — COMMENTED** [review]
Thanks for contributing to Allo. The newly added modules are very helpful!

File: allo/frontend/library.py (hunk 1 above)
Comment (ZRQ0410) [reply allo/frontend/library.py:50]: Added new comment and reference

File: allo/frontend/pytorch.py (hunk 2 above)
Comment (ZRQ0410) [reply allo/frontend/pytorch.py]: Done
